### ISSUE TITLE
Enhance address change functionality

### ISSUE THREAD

**Issue by NoahAndrews** [REVrobotics/node-rhsplib#50]
Changing the Expansion Hub address should optionally (defaulting to true) perform discovery to check that no other devices on the bus have the same address. Additionally, `ParentExpansionHub` should have a `discoverChildren(openNewlyFoundChildren: boolean = true)` method that allows the application to prevent the user from selecting an address that's already being used on the bus.

**Comment by NoahAndrews** [comment]
I agree with the approach, but I don't love the names. Maybe "findAddressesOfPotentialChildren" and "openAllChildren"? The first one is tricky, since we don't want to give the impression that we're just listing the addresses of the children that we're currently communicating with. The doc comment will need to spell out what's happening clearly.

**Comment by LandryNorris** [comment]
findAddressesOfUsbConnectedChildren? It's quite long, but indicates what the devices are.

**Comment by NoahAndrews** [comment]
> findAddressesOfUsbConnectedChildren? It's quite long, but indicates what the devices are. Maybe devices is a better name than children in this case?

It's RS-485 not USB, but otherwise I like it.

The other thing to consider is the Control Hub, where we'll need to support doing this for USB and RS-485 children.

**Comment by LandryNorris** [comment]
How about `findAddressesOfAllChildrenConnectedToThisExpansionOrControlHubWhichMayBeConnectedViaRs485OrUsb()`?
